table/block.cc

```diff
@@ -505,9 +532,81 @@ bool IndexBlockIter::PrefixSeek(const Slice& target, uint32_t* index) {
   }
 }
 
+// NOTE: in hash seek, if the key is not found in the restart intervals,
+// the iterator will simply be set as "invalid", rather than returning
+// the key that is just pass the target key.
+void DataBlockIter::HashSeek(const Slice& target) {
+  assert(data_block_hash_index_);
+  Slice user_key = ExtractUserKey(target);
+  uint8_t entry = data_block_hash_index_->Seek(user_key);
+
+  if (entry == kNoEntry) {
+    current_ = restarts_;  // not found, Invalidate the iterator
```

Comment:
In this case, the iterator will be tag as Invalid(), which means the seek key is larger than the largest key in the block. In this case, Get() will go to the next data block and SeekToFirst(). I don't think this is what we want. We should not do this.